server/src/jobs/fiabilisation/uai-siret/build.rules.ts

```diff
@@ -0,0 +1,405 @@
+      const organismesRespOrRespFormateurForSiretTdb = await organismesReferentielDb()
+        .find({
+          siret: currentMultipleSiretCouple.siret,
+          $or: [
+            { nature: NATURE_ORGANISME_DE_FORMATION.RESPONSABLE_FORMATEUR },
+            { nature: NATURE_ORGANISME_DE_FORMATION.RESPONSABLE },
+          ],
+        })
+        .toArray();
+
+      // si SIRET unique au niveau du référentiel alors on y associe l'UAI présent dans référentiel
+      if (organismesRespOrRespFormateurForSiretTdb.length === 1) {
+        // Phase 1. Recherche dans les relations
+        const relationsReponsableWithSiret =
+          organismesRespOrRespFormateurForSiretTdb[0]?.relations?.filter((item) => item.siret) || [];
+
+        if (relationsReponsableWithSiret) {
+          // Pour chacune des relations du responsable si l'UAI Match alors on peut fiabiliser en remplacant le siret de la relation
+          for (const currentRelation of relationsReponsableWithSiret) {
+            // On vérifie si la relation match sur l'UAI
+            const relationMatchingUai = await organismesReferentielDb().countDocuments({
+              siret: currentRelation.siret as string,
+              uai: currentMultipleSiretCouple.uai,
+            });
+
+            // SI Match Relation UAI alors on ajoute le couple à fiabiliser avec cet UAI
+            if (relationMatchingUai > 0) {
+              await fiabilisationUaiSiretDb().updateOne(
+                { uai: coupleUaiSiretTdbToCheck.uai, siret: coupleUaiSiretTdbToCheck.siret },
+                {
+                  $set: {
+                    uai_fiable: currentMultipleSiretCouple.uai,
+                    siret_fiable: coupleUaiSiretTdbToCheck.siret,
+                    type: STATUT_FIABILISATION_COUPLES_UAI_SIRET.A_FIABILISER,
+                  },
+                },
+                { upsert: true }
+              );
+
+              return true;
+            }
+          }
+        }
+
+        // Phase 2. Recherche dans les lieux de formation
+        const lieuFormationFiableMatchUai = organismesRespOrRespFormateurForSiretTdb[0].lieux_de_formation.find(
+          (item) => item.uai === currentMultipleSiretCouple.uai && item.uai_fiable
+        );
+
+        // Si l'UAI Match sur un des lieux fiables alors on peut fiabiliser tel quel en marquant que c'est un couple de lieu de formation
+        if (lieuFormationFiableMatchUai) {
+          await fiabilisationUaiSiretDb().updateOne(
+            { uai: coupleUaiSiretTdbToCheck.uai, siret: coupleUaiSiretTdbToCheck.siret },
+            {
+              $set: {
+                uai_fiable: coupleUaiSiretTdbToCheck.uai,
+                siret_fiable: coupleUaiSiretTdbToCheck.siret,
+                type: STATUT_FIABILISATION_COUPLES_UAI_SIRET.A_FIABILISER,
+              },
+            },
+            { upsert: true }
+          );
+
+          return true;
```

Comment:
Attention, le return ici ne sort pas de la fonction, mais du PromisePool. D'ailleurs, tu ne veux peut-être pas l'utiliser pour ce cas mais plutôt faire les vérifications séquentiellement ?

Reply:
Si la fonction de la rule renvoie true, on devrait normalement return dans la fonction parente (buildFiabilisationCoupleForTdbCouple) et sortir.

Reply:
Aprés verif il faut bien qu'on sorte dès qu'on trouve une entrée à fiabiliser, donc le remplacement du PromisePool par for of fait le taf je dirais (on a une tte petite volumétrie)